### PR TITLE
Created BadMetrics table and access primitives

### PR DESCRIPTION
**Pull request by GeorgeJahad**
What:
Table for storing bad metrics and the read/write primitives to update it.

Why:
To be able to check incoming metrics against a list of know bad metrics.

How:
A Cassandra table with a Locator per row, and a timestamp for the single column

### CODE REVIEW

File: blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/CassandraModel.java [hunk 1]
```diff
@@ -67,6 +67,10 @@
             LongSerializer.get(),
             SlotStateSerializer.get());
 
+    public static final ColumnFamily<Locator, Long> CF_METRICS_BAD_METRICS = new ColumnFamily<Locator, Long>("metrics_bad_metrics",
```
Comment (tilogaat) [inline blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/CassandraModel.java]: should we make it clear that these are bad enum metrics?

File: blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/AstyanaxReader.java [hunk 2]
```diff
@@ -520,4 +523,40 @@ private Points getPointsFromColumns(ColumnList<Long> columnList, RollupType roll
...
+                return true;
+            }
+        };
+
+        ColumnFamily CF = CassandraModel.CF_METRICS_BAD_METRICS;
+        Timer.Context ctx = Instrumentation.getBatchReadTimerContext(CF);
+        try {
+            new AllRowsReader.Builder<Locator, Long>(keyspace, CassandraModel.CF_METRICS_BAD_METRICS)
+                    .withColumnRange(null, null, false, 0)
+                    .withPartitioner(null) // this will use keyspace's partitioner
```
Comment (tilogaat) [inline blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/AstyanaxReader.java]: Why do we need the partitioner?

**Comment by VinnyQ** [comment]
:+1:  lgtm.  

You will be using the writeBadMetrics() method during rollup to add the locator (tenantId:metric_name) to this table if num distinct enum values for that metric are > threshold?

And getBadMetrics() method during ingestion to see if the metric is flagged as a bad metric?

File: blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/CassandraModel.java (hunk 1 above)
Comment (GeorgeJahad) [inline blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/CassandraModel.java]: will do.

File: blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/AstyanaxReader.java (hunk 2 above)
Comment (GeorgeJahad) [inline blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/AstyanaxReader.java]: we don't.  i'll remove

**Comment by GeorgeJahad** [comment]
Yes, writeBadMetric() and getBadMetrics() are to be used as you describe.

**Comment by tilogaat** [comment]
+1

**Pull request merged**